python/pyarrow/pandas_compat.py

```diff
@@ -450,6 +455,18 @@ def table_to_blockmanager(options, table, memory_pool, nthreads=1):
         index = pd.RangeIndex(row_count)
 
     column_strings = [x.name for x in block_table.itercolumns()]
+    if 'columns_metadata' in locals() and columns_metadata is not None:
```

Comment:
I'm not a fan of using `locals()` though I don't know exactly why. I'm going to remove this and make `columns_metadata` always None even if there is no pandas metadata